### PR TITLE
Adding support for group issue boards

### PR DESCRIPTION
**Pull request by pmwebster**
I had to add this functionality for something I'm playing with and thought you might like to pull it in :) Thanks for making this library! Feel free to let me know if I can be of any other use with it, always happy to help when time permits.

### CODE REVIEW

File: group_boards.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// GroupIssueBoardsService handles communication with the group issue board related
```
Comment (svanharmelen) [inline group_boards.go]: Please try to keep comment lines to max 80 chars (and code lines to max 100 chars, except for function declarations)

File: group_boards.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+// GroupIssueBoard represents a GitLab group issue board.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/group_boards.html
+type GroupIssueBoard struct {
+	ID        int          `json:"id"`
+	Name      string       `json:"name"`
+	Group     *Group       `json:"group"`
+	Milestone *Milestone   `json:"milestone"`
+	Lists     []*BoardList `json:"lists"`
+}
```
Comment (svanharmelen) [inline group_boards.go]: In the examples in the docs I don't see the `name` and `milestone` fields. Are you sure they are returned by the API?

File: group_boards.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+func (b GroupIssueBoard) String() string {
+	return Stringify(b)
+}
+
+// ListGroupIssueBoardsOptions represents the available ListGroupIssueBoards() options.
```
Comment (svanharmelen) [inline group_boards.go]: Please keep comments to max 80 chars

File: group_boards.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+// GetGroupIssueBoardListsOptions represents the available GetGroupIssueBoardLists() options.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/group_boards.html#list-board-lists
+type GetGroupIssueBoardListsOptions ListOptions
+
+// GetGroupIssueBoardLists gets a list of the issue board's lists. Does not include
+// backlog and closed lists.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/group_boards.html#list-board-lists
+func (s *GroupIssueBoardsService) GetGroupIssueBoardLists(gid interface{}, board int, opt *GetGroupIssueBoardListsOptions, options ...OptionFunc) ([]*BoardList, *Response, error) {
```
Comment (svanharmelen) [inline group_boards.go]: I would call this `ListGroupIssueBoardLists()` to match the docs...

File: group_boards.go (hunk 2 above)
Comment (pmwebster) [reply group_boards.go]: Interesting -- they're not in the ce api, but they definitely are in the ee api which we use in house -- https://docs.gitlab.com/ee/api/group_boards.html#group-board

Comment (svanharmelen) [reply group_boards.go]: Check! Didn't check those, but then it's fine 👍

**Comment by pmwebster** [comment]
Thanks for the feedback! I cleaned up all of the comments (including the ones you mentioned) and the function name change is in place.

File: group_boards.go [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+	var gbl []*BoardList
+	resp, err := s.client.Do(req, gbl)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return gbl, resp, err
+}
+
+// DeleteGroupIssueBoardList soft deletes a group issue board list. Only for admins and
```
Comment (svanharmelen) [inline group_boards.go]: Sorry to be nitting here, but seems you forgot this line... 

As well as some lines that contain that the doc links (earlier in this file). The links are of course always a single line, but then they could be written as:

```
// GitLab API docs:
// https://docs.gitlab.com/ce/api/group_boards.html#delete-a-board-list
```

**Comment by pmwebster** [comment]
You're totally right, sorry about that!

So far as the doc links, I realized that when I started throwing together the other libs I made PRs for, forgot to update this one. Should be good now.

**Comment by svanharmelen** [comment]
Thank for putting up with me :wink: Thanks once again!

**Pull request merged**